### ISSUE TITLE
tidal-1.9.2 segfaults on 32-bit/i386 with startTidal(...)

### ISSUE THREAD

**Issue by cleary** [tidalcycles/Tidal#957]
tested on devuan 4.0/i386 (debian-based)

```
cabal-install version 3.0.0.0
compiled using version 3.0.1.0 of the Cabal library
```

Confirmed using the BootTidal.hs loaded directly in ghci - if I comment the `startTidal(...)` line, the file loads cleanly to the end

If I include the `startTidal(...)` line, I get the following:
```
GHCi, version 8.8.4: https://www.haskell.org/ghc/  :? for help
Loaded package environment from /home/bernie/.ghc/i386-linux-8.8.4/environments/default
Prelude> :script BootTidal.hs
[TidalCycles version 1.9.2]
Installed in /home/bernie/.cabal/store/ghc-8.8.4/tidal-1.9.2-aa64e723f878646ed318ab6b17f3cd857dfb5f63d554d046ddd2930bf1e986a0/share
Listening for external controls on 127.0.0.1:6010
Segmentation fault
```

If I revert to tidal-1.8.0, it works as expected

**Comment by Zalastax** [comment]
Hi @cleary, Tidal 1.9.x interops with C++ to use Ableton Link. C++ interop is not so common in Haskell, so the bug might be in GHC. There's a possibility that the issue you are facing is solved in later versions of GHC - this is the case for Windows. Would you be willing to try upgrading GHC and report back?

**Comment by cleary** [comment]
Yep can do, I'll get back to you in a couple of days 🙂

**Comment by cleary** [comment]
OK I have a tidal-link build failure on ghc 9.4.2 (attached). 
I don't understand the message particularly well, so I'll leave it here in the meantime just in case you want to have a look - there are warnings at the start about ghc-9.4.2 not being supported by cabal 3.0.1.0 so I'll look at upgrading cabal first.
 
[tidal-link-build-log-error.txt](https://github.com/tidalcycles/Tidal/files/9823538/tidal-link-build-log-error.txt)

**Comment by Zalastax** [comment]
That's a bummer!

Does i386 mean that you have some unusual hardware? There is some risk that there is a bug in GHC. But finding a minimal case to report as a bug can take a fair amount of dedication.

Some things to try:
* Start ghci with `ghci -v3` to see if we can get some more details.
* Run the hello function from tidal-link to see how far it gets https://github.com/tidalcycles/Tidal/blob/main/tidal-link/src/hs/Main.hs
* Download https://github.com/Zalastax/ghc-bug-20918 to see if exception handling works as expected

**Comment by cleary** [comment]
> That's a bummer!
> 
> Does i386 mean that you have some unusual hardware? There is some risk that there is a bug in GHC. But finding a minimal case to report as a bug can take a fair amount of dedication.

No, it's a bog standard old Toshiba Satellite laptop with an x86_64 pentium cpu, just with an i386 install for the purposes of testing

> Some things to try:
> 
>     * Start ghci with `ghci -v3` to see if we can get some more details.
> 
>     * Run the hello function from tidal-link to see how far it gets https://github.com/tidalcycles/Tidal/blob/main/tidal-link/src/hs/Main.hs
> 
>     * Download https://github.com/Zalastax/ghc-bug-20918 to see if exception handling works as expected

I'll give this a look over the next few days - thanks :)

**Comment by Zalastax** [comment]
Thanks a lot @cleary! If you get stuck, I can try booting a virtual machine to see how it behaves for me,

**Comment by sss-create** [comment]
Is this still a problem or did we stop supporting i386 at all?

**Comment by sss-create** [comment]
closing this because of inactivity. feel free to reopen.:)
